src/Symfony/Component/Messenger/Transport/Enhancers/MemoryLimitReceiver.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Enhancers;
+
+use Symfony\Component\Messenger\Transport\ReceiverInterface;
+
+/**
+ * @author Simon Delicata <simon.delicata@free.fr>
+ */
+class MemoryLimitReceiver implements ReceiverInterface
+{
+    private $decoratedReceiver;
+    private $memoryLimit;
+
+    public function __construct(ReceiverInterface $decoratedReceiver, string $memoryLimit)
+    {
+        $this->decoratedReceiver = $decoratedReceiver;
+        $this->memoryLimit = $this->convertToOctets($memoryLimit);
+    }
+
+    public function receive(callable $handler): void
+    {
+        $this->decoratedReceiver->receive(function ($message) use ($handler) {
+            $handler($message);
+
+            if (\memory_get_usage() >= $this->memoryLimit) {
+                $this->stop();
+            }
+        });
+    }
+
+    public function stop(): void
+    {
+        $this->decoratedReceiver->stop();
+    }
+
+    private function convertToOctets(string $size): int
+    {
+        if (\preg_match('/^(\d+)(.)$/', $size, $matches)) {
```

Comment:
I'd be stricter than `.`. Maybe just `(G|M|K)` and throw an exception if nothing is recognized to prevent any issue.

Reply:
Done